### PR TITLE
[chore]: dont set path for go in validate

### PR DESCRIPTION
**Pull request by sakshi1215**
Signed-off-by: sakshi1215 <sakshi.patle121@gmail.com>

**Description:** 

Removing setting path in validate

**Link to tracking Issue:** fixes https://github.com/open-telemetry/opentelemetry-collector/issues/6030

### CODE REVIEW

**Comment by sakshi1215** [comment]
@dmitryax can you please trigger the CI on this? Also, I am unable to sign CLA as individual contributor.

**Comment by jpkrohling** [comment]
I could trigger this, but I don't think this is the right solution -- the mismatch is that the name "Validate" implies a function that does not change the state. Setting the path is indeed correct, so perhaps we just need to change the name of the function?

**Comment by sakshi1215** [comment]
Oh I might have misunderstood the context of the same, let me make the
changes you suggested.

File: CHANGELOG.md [hunk 1]
```diff
@@ -52,7 +52,7 @@
 ### 🧰 Bug fixes 🧰
 
 - Fixed bug where `telemetryInitializer` is not cleaned up when `newService` errors (#6239)
-
+- (Builder)Remove setting of the Go executable from Validate fn (#6273)
```
Comment (dmitryax) [inline CHANGELOG.md:55]: This is an internal refactoring. Doesn't need a changelog entry

Comment (kushthedude) [reply CHANGELOG.md:55]: Fixing

Comment (sakshi1215) [reply CHANGELOG.md]: @dmitryax can you please check again?

File: cmd/builder/internal/builder/config.go [hunk 2]
```diff
@@ -88,8 +88,8 @@ func NewDefaultConfig() Config {
 	}
 }
 
-// Validate checks whether the current configuration is valid
-func (c *Config) Validate() error {
+// ValidateAndSetGoPath checks whether the current configuration is valid
```
Comment (dmitryax) [inline cmd/builder/internal/builder/config.go:91]:
```suggestion
// ValidateAndSetGoPath checks whether the current configuration is valid and sets go path
```

**Review by dmitryax — APPROVED** [review]

**Comment by sakshi1215** [comment]
Can we merge this @dmitryax ?

**Comment by bogdandrutu** [comment]
@jpkrohling are you ok with this?

File: cmd/builder/internal/builder/config.go [hunk 3]
```diff
@@ -88,8 +88,8 @@ func NewDefaultConfig() Config {
 	}
 }
 
-// Validate checks whether the current configuration is valid
-func (c *Config) Validate() error {
+// ValidateAndSetGoPath checks whether the current configuration is valid and sets go path
+func (c *Config) ValidateAndSetGoPath() error {
```
Comment (jpkrohling) [inline cmd/builder/internal/builder/config.go:92]: I think I would prefer to split this function into two:

- SetGoPath
- Validate

The current callers of `Validate` will then need to call both.

Comment (sakshi1215) [reply cmd/builder/internal/builder/config.go]: Done @jpkrohling

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please fix tests.

**Comment by sakshi1215** [comment]
@bogdandrutu Please review this again, I have fixed tests

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**